### PR TITLE
Remove unnecessary code when reading CacheFile

### PR DESCRIPTION
**Pull request by tlrx**
The code that is removed in pull request has been introduced when Lucene was upgrade to 8.6.0, in which `BufferedIndexInput` read method is changed to use a `ByteBuffer` instead of a fixed buffer.

The range to read here is computed from the ByteBuffer position and remaining bytes so it should never exceed the ByteBuffer's remaining bytes. Also, Lucene guarantees that the `ByteBuffer` [never read more bytes than the file total length](https://github.com/apache/lucene-solr/blob/branch_8_8/lucene/core/src/java/org/apache/lucene/store/BufferedIndexInput.java#L150-L151) so I think we can safely remove this unnecessary code.

### CODE REVIEW

File: x-pack/plugin/searchable-snapshots/src/main/java/org/elasticsearch/index/store/cache/CachedBlobContainerIndexInput.java [hunk 1]
```diff
@@ -290,19 +290,13 @@ protected void readInternal(ByteBuffer b) throws IOException {
...
-                    duplicate.limit(duplicate.position() + toIntBytes(rangeToRead.v2() - rangeToRead.v1()));
-                    read = readCacheFile(channel, position, duplicate);
-                    assert duplicate.position() <= b.limit();
-                    b.position(duplicate.position());
-                } else {
-                    read = readCacheFile(channel, position, b);
-                }
-                return read;
-            }, this::writeCacheFile, directory.cacheFetchAsyncExecutor());
+            final Future<Integer> populateCacheFuture = cacheFile.populateAndRead(
```
Comment (original-brownbear) [inline x-pack/plugin/searchable-snapshots/src/main/java/org/elasticsearch/index/store/cache/CachedBlobContainerIndexInput.java:293]: Maybe 

```
assert rangeToRead.v2() - rangeToRead.v1()) >= b.remaining()
```

here ?

**Review by original-brownbear — APPROVED** [review]
LGTM

File: x-pack/plugin/searchable-snapshots/src/main/java/org/elasticsearch/index/store/cache/CachedBlobContainerIndexInput.java (hunk 1 above)
Comment (tlrx) [reply x-pack/plugin/searchable-snapshots/src/main/java/org/elasticsearch/index/store/cache/CachedBlobContainerIndexInput.java:293]: I think we can assert `rangeToRead.v2() - rangeToRead.v1() == b.remaining()`

**Comment by tlrx** [comment]
Thanks Armin

**Pull request merged**